From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander <protonull@protonmail.com>
Date: Wed, 5 Jun 2024 15:18:50 +0100
Subject: [PATCH] Make CivMC specific


diff --git a/civmc b/civmc
deleted file mode 100644
index e69de29bb2d1d6434b8b29ae775ad8c2e48c5391..0000000000000000000000000000000000000000
diff --git a/common/src/main/java/sh/okx/civmodern/common/radar/Radar.java b/common/src/main/java/sh/okx/civmodern/common/radar/Radar.java
index 2f1653ef51b0920dcb8a0d6769559e97feb6b1dd..610ad6730b13fd01f8b73558cd5fca98ea39d2c8 100644
--- a/common/src/main/java/sh/okx/civmodern/common/radar/Radar.java
+++ b/common/src/main/java/sh/okx/civmodern/common/radar/Radar.java
@@ -45,7 +45,6 @@ import sh.okx.civmodern.common.events.ClientTickEvent;
 import sh.okx.civmodern.common.events.EventBus;
 import sh.okx.civmodern.common.events.PostRenderGameOverlayEvent;
 
-import java.net.URL;
 import java.util.Collections;
 import java.util.HashSet;
 import java.util.Random;
@@ -55,18 +54,6 @@ import java.util.UUID;
 import static org.lwjgl.opengl.GL11.*;
 
 public class Radar {
-
-    private static boolean hideY;
-
-    static {
-        URL resource = Radar.class.getResource("/civmc");
-        if (resource != null) {
-            hideY = true;
-        } else {
-            hideY = false;
-        }
-    }
-
     private final EventBus eventBus;
     private final ColourProvider colourProvider;
     private final CivMapConfig config;
@@ -96,7 +83,7 @@ public class Radar {
     }
 
     private boolean hideY() {
-        return hideY;
+        return true;
     }
 
     @Subscribe
diff --git a/fabric/build.gradle b/fabric/build.gradle
index 83fec77b3e4844561496b42580cf686b27f1fff7..18be897a8fbbf9fe919a2979d346332941a649ef 100644
--- a/fabric/build.gradle
+++ b/fabric/build.gradle
@@ -78,16 +78,6 @@ publishing {
     }
 }
 
-import net.fabricmc.loom.task.RemapJarTask
-
-tasks.register("remapCivMc", RemapJarTask) {
-    inputFile.set shadowJar.archiveFile
-    dependsOn shadowJar
-    archiveClassifier = "civmc-fabric"
-    from sourceSets.main.output
-    from "../civmc"
-}
-
 tasks.register("cleanJar", Delete) {
     delete fileTree("../dist") {
         include "*-fabric.jar"
@@ -96,7 +86,7 @@ tasks.register("cleanJar", Delete) {
 
 tasks.register("copyJar", Copy) {
     dependsOn cleanJar
-    from remapJar, remapCivMc
+    from remapJar
     into "../dist"
 }
 
diff --git a/forge/build.gradle b/forge/build.gradle
index 94d7fc77147ded7538d6205c8a9d3ba7a0de8841..255dfedc6edb059cf59c1f6bd75ce352405d9a98 100644
--- a/forge/build.gradle
+++ b/forge/build.gradle
@@ -86,16 +86,6 @@ publishing {
     }
 }
 
-import net.fabricmc.loom.task.RemapJarTask
-
-tasks.register("remapCivMc", RemapJarTask) {
-    inputFile.set shadowJar.archiveFile
-    dependsOn shadowJar
-    archiveClassifier = "civmc-forge"
-    from sourceSets.main.output
-    from "../civmc"
-}
-
 tasks.register("cleanJar", Delete) {
     delete fileTree("../dist") {
         include "*-forge.jar"
@@ -104,7 +94,7 @@ tasks.register("cleanJar", Delete) {
 
 tasks.register("copyJar", Copy) {
     dependsOn cleanJar
-    from remapJar, remapCivMc
+    from remapJar
     into "../dist"
 }
 
