From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander <protonull@protonmail.com>
Date: Wed, 5 Jun 2024 15:07:19 +0100
Subject: [PATCH] Add ModMenu and Forge mod-list support


diff --git a/build.gradle b/build.gradle
index 27355a58bf4df4c9aadd0f7bb70cdb8e4ad871f0..c45fdcad2f11098029583978012a9ce45fbe8fca 100644
--- a/build.gradle
+++ b/build.gradle
@@ -40,6 +40,12 @@ allprojects {
             name = "ParchmentMC"
             url = "https://maven.parchmentmc.org"
         }
+        maven {
+            url "https://api.modrinth.com/maven"
+            content {
+                includeGroup "maven.modrinth"
+            }
+        }
     }
 
     tasks.withType(JavaCompile).configureEach {
diff --git a/common/src/main/java/sh/okx/civmodern/common/AbstractCivModernMod.java b/common/src/main/java/sh/okx/civmodern/common/AbstractCivModernMod.java
index df92f08652acb07a311953768786fbc8d49ea40f..3659b7dfdefffdddd535eb4654068b63b91aa1d2 100644
--- a/common/src/main/java/sh/okx/civmodern/common/AbstractCivModernMod.java
+++ b/common/src/main/java/sh/okx/civmodern/common/AbstractCivModernMod.java
@@ -13,6 +13,7 @@ import java.util.Properties;
 import net.minecraft.client.KeyMapping;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.Options;
+import net.minecraft.client.gui.screens.Screen;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.jetbrains.annotations.NotNull;
@@ -158,6 +159,12 @@ public abstract class AbstractCivModernMod {
         return colourProvider;
     }
 
+    public @NotNull Screen newConfigGui(
+        final Screen previousScreen
+    ) {
+        return new MainConfigScreen(this, this.config);
+    }
+
     public static AbstractCivModernMod getInstance() {
         return INSTANCE;
     }
diff --git a/common/src/main/java/sh/okx/civmodern/common/gui/screen/CompactedConfigScreen.java b/common/src/main/java/sh/okx/civmodern/common/gui/screen/CompactedConfigScreen.java
index d274897284faa8b18cb322c9c4df3fec059c3809..e656c6de58d64e6fc7c948b48fbbf3279d42e7a7 100644
--- a/common/src/main/java/sh/okx/civmodern/common/gui/screen/CompactedConfigScreen.java
+++ b/common/src/main/java/sh/okx/civmodern/common/gui/screen/CompactedConfigScreen.java
@@ -13,6 +13,7 @@ import net.minecraft.network.chat.CommonComponents;
 import net.minecraft.network.chat.Component;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.world.item.ItemStack;
+import org.lwjgl.glfw.GLFW;
 import sh.okx.civmodern.common.AbstractCivModernMod;
 import sh.okx.civmodern.common.CivMapConfig;
 import sh.okx.civmodern.common.ColourProvider;
@@ -113,14 +114,17 @@ public class CompactedConfigScreen extends Screen {
     }
 
     @Override
-    public boolean mouseClicked(double mouseX, double mouseY, int button) {
-        LocalPlayer player = Minecraft.getInstance().player;
-        if (isCursorOverItem((int) mouseX, (int) mouseY) && button == 0 && player.isCreative()) {
+    public boolean mouseClicked(
+        final double mouseX,
+        final double mouseY,
+        final int button
+    ) {
+        final LocalPlayer player = Minecraft.getInstance().player;
+        if (player != null && isCursorOverItem((int) mouseX, (int) mouseY) && button == GLFW.GLFW_MOUSE_BUTTON_1 && player.isCreative()) {
             player.addItem(ITEM.copy());
             return true;
-        } else {
-            return super.mouseClicked(mouseX, mouseY, button);
         }
+        return super.mouseClicked(mouseX, mouseY, button);
     }
 
     @Override
diff --git a/fabric/build.gradle b/fabric/build.gradle
index 0386193a83d2903bfe4aed83f6b7212544ed3c83..18be897a8fbbf9fe919a2979d346332941a649ef 100644
--- a/fabric/build.gradle
+++ b/fabric/build.gradle
@@ -21,6 +21,9 @@ dependencies {
 
     common(project(path: ":common", configuration: "namedElements")) { transitive = false }
     shadowCommon(project(path: ":common", configuration: "transformProductionFabric")) { transitive = false }
+
+    // https://modrinth.com/mod/modmenu/version/7.2.2
+    modCompileOnly("maven.modrinth:modmenu:lEkperf6")
 }
 
 processResources {
diff --git a/fabric/src/main/java/sh/okx/civmodern/fabric/integrations/modmenu/ModMenuIntegration.java b/fabric/src/main/java/sh/okx/civmodern/fabric/integrations/modmenu/ModMenuIntegration.java
new file mode 100644
index 0000000000000000000000000000000000000000..5e3ce45c7957a66f3885ec7593c829549b498a62
--- /dev/null
+++ b/fabric/src/main/java/sh/okx/civmodern/fabric/integrations/modmenu/ModMenuIntegration.java
@@ -0,0 +1,13 @@
+package sh.okx.civmodern.fabric.integrations.modmenu;
+
+import com.terraformersmc.modmenu.api.ConfigScreenFactory;
+import com.terraformersmc.modmenu.api.ModMenuApi;
+import org.jetbrains.annotations.NotNull;
+import sh.okx.civmodern.common.AbstractCivModernMod;
+
+public final class ModMenuIntegration implements ModMenuApi {
+    @Override
+    public @NotNull ConfigScreenFactory<?> getModConfigScreenFactory() {
+        return (previousScreen) -> AbstractCivModernMod.getInstance().newConfigGui(previousScreen);
+    }
+}
diff --git a/fabric/src/main/resources/fabric.mod.json b/fabric/src/main/resources/fabric.mod.json
index 95f40c181953532aa0bd73774675bdd09a10cbeb..03dec00db119c15aa7d4cd0952ede1dcdfc42c9e 100644
--- a/fabric/src/main/resources/fabric.mod.json
+++ b/fabric/src/main/resources/fabric.mod.json
@@ -17,6 +17,9 @@
     "entrypoints": {
         "client": [
             "sh.okx.civmodern.fabric.FabricCivModernBootstrap"
+        ],
+        "modmenu": [
+            "sh.okx.civmodern.fabric.integrations.modmenu.ModMenuIntegration"
         ]
     },
     "depends": {
diff --git a/forge/src/main/java/sh/okx/civmodern/forge/ForgeCivModernBootstrap.java b/forge/src/main/java/sh/okx/civmodern/forge/ForgeCivModernBootstrap.java
index a0959a521c06895534954c5e7a2aa3cdb8d00378..f9d89db3e39e511edbfe072df3bf8b87a1bf1c1d 100644
--- a/forge/src/main/java/sh/okx/civmodern/forge/ForgeCivModernBootstrap.java
+++ b/forge/src/main/java/sh/okx/civmodern/forge/ForgeCivModernBootstrap.java
@@ -1,15 +1,18 @@
 package sh.okx.civmodern.forge;
 
+import net.minecraftforge.client.ConfigScreenHandler;
 import net.minecraftforge.client.event.RenderGuiEvent;
 import net.minecraftforge.client.event.RenderLevelStageEvent;
 import net.minecraftforge.common.MinecraftForge;
 import net.minecraftforge.event.TickEvent;
 import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.ModLoadingContext;
 import net.minecraftforge.fml.common.Mod;
 import net.minecraftforge.fml.event.lifecycle.FMLClientSetupEvent;
 import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
+import sh.okx.civmodern.common.AbstractCivModernMod;
 import sh.okx.civmodern.common.events.ClientTickEvent;
 import sh.okx.civmodern.common.events.PostRenderGameOverlayEvent;
 import sh.okx.civmodern.common.events.WorldRenderEvent;
@@ -29,6 +32,14 @@ public class ForgeCivModernBootstrap {
     public void clientSetup(FMLClientSetupEvent event) {
         this.mod.init();
         this.mod.enable();
+
+        // Register hook for Forge's native mod list
+        ModLoadingContext.get().registerExtensionPoint(
+            ConfigScreenHandler.ConfigScreenFactory.class,
+            () -> new ConfigScreenHandler.ConfigScreenFactory(
+                (minecraft, previousScreen) -> AbstractCivModernMod.getInstance().newConfigGui(previousScreen)
+            )
+        );
     }
 
     @SubscribeEvent
